brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java

```diff
@@ -130,29 +153,34 @@ public ToscaNodeToEntityConverter setIndexedNodeTemplate(IndexedArtifactToscaEle
 
         configurePropertiesForSpec(spec, nodeTemplate);
 
+        configureRuntimeEnvironment(spec);
+
         // If the entity spec is of type VanillaSoftwareProcess, we assume that it's running. The operations should
         // then take care of setting up the correct scripts.
         if (spec.getType().isAssignableFrom(VanillaSoftwareProcess.class)) {
             spec.configure(VanillaSoftwareProcess.LAUNCH_COMMAND, "true");
             spec.configure(VanillaSoftwareProcess.STOP_COMMAND, "true");
             spec.configure(VanillaSoftwareProcess.CHECK_RUNNING_COMMAND, "true");
+            Map<String, Object> shellEnvironment = MutableMap.of();
+            for (Map.Entry<String, ?> entry : nodeTemplate.getProperties().entrySet()) {
+                if (entry.getValue() instanceof ScalarPropertyValue) {
+                    shellEnvironment.put(entry.getKey().toUpperCase(), ((ScalarPropertyValue) entry.getValue()).getValue());
```

Comment:
This looks like it's actually going to be a bit more difficult as we need to get it from the inputs instead. I've pushed a 'temp commit' branch with a spike of how it's done (for backup purposes) and will do it properly tomorrow